lib/puppet/settings.rb

```diff
@@ -155,6 +160,27 @@ def initialize
     @config_file_parser = Puppet::Settings::ConfigFile.new(@translate)
   end
 
+  # Update the cache with content to be used by future threads
+  # Do not use if multiple threads already exist
+  def unsafe_snapshot_cache
```

Comment:
Is this a helper for testing?

Reply:
It's only called in the testing, but it's meant to mark the cache as shareable with all future threads, and probably should be called by Puppetserver at some point?.

Without this function, anytime we start a new `Thread`, the `@cache` will be a new empty map, per the block passed to the `Puppet::ThreadLocal.new` assignment of the `@cache` in initialization. It wasn't apparent to me if this new empty map was desireable, or that marking the `@cache` with content to be used by future threads was. The former seems to be very shortlived for a cache, so I sorta assumed that the latter was going to be necessary.